packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

```diff
@@ -849,6 +852,7 @@ contract Deploy is Deployer {
 
     /// @notice Deploy the AnchorStateRegistry
     function deployAnchorStateRegistry() public broadcast returns (address addr_) {
+        console.log("Deploying AnchorStateRegistry implementation");
         IAnchorStateRegistry anchorStateRegistry =
             IAnchorStateRegistry(_deploy("AnchorStateRegistry", abi.encode(mustGetAddress("DisputeGameFactoryProxy"))));
```

Comment:
> Also the ASR's implementation was unnecessarily initialized with a proxy address for a constructor
argument.

It looks like this wasn't changed? Unsure if you wanted to fix that in this PR or not, but the changes here look good, so approving this PR

Reply:
That was incorrect, the constructor arg is needed because the ASR is not MCP ready and places the DGF address as an immutable. The real fix was moving both the ASR proxy and impl into `deployOpChain`.